### PR TITLE
feat: [CDS-40903]: move all yaml parsing from yaml to js-yaml

### PR DESCRIPTION
**Pull request by vkbansal**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/10-common/utils/YamlHelperMethods.ts [hunk 1]
```diff
@@ -5,21 +5,24 @@
...
 
-import { Document } from 'yaml'
-import type { Options } from 'yaml'
-import { load } from 'js-yaml'
+/* disable lint in this file as we are exporting wrappers */
+
+// eslint-disable-next-line no-restricted-imports
+import type { DumpOptions } from 'js-yaml'
+// eslint-disable-next-line no-restricted-imports
+import { load, dump } from 'js-yaml'
```
Comment (knagurski) [inline src/modules/10-common/utils/YamlHelperMethods.ts:13]: Could these be combined?
```suggestion
import { load, dump, DumpOptions } from 'js-yaml'
```

Comment (vkbansal) [reply src/modules/10-common/utils/YamlHelperMethods.ts:13]: I prefer to keep type imports separate for better readability

File: scripts/generate-eslint-config-for-imports.mjs [hunk 2]
```diff
@@ -27,27 +27,54 @@ for (const { dirName, moduleName } of flattenedLayers) {
...
-    ]
+  const customConfigFile = path.join(modulePath, 'custom.eslintrc.yml')
+
+  if (fs.existsSync(customConfigFile)) {
+    const customConfig = yaml.load(fs.readFileSync(customConfigFile), 'utf8')
+    console.log('customConfig', customConfig)
+
+    if (customConfig.rules) {
+      Object.entries(customConfig.rules).forEach(([rule, ruleConfig]) => {
+        console.log('rule', rule)
```
Comment (bhavya-sinha) [inline scripts/generate-eslint-config-for-imports.mjs:54]: console log?

Comment (vkbansal) [reply scripts/generate-eslint-config-for-imports.mjs:54]: Will remove it in next PR

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx [hunk 3]
```diff
@@ -392,7 +393,7 @@ export function OverlayInputSetForm({
         initialStoreMetadata
       })
       response = isEdit
-        ? await updateOverlayInputSet(requestData, {
+        ? await updateOverlayInputSet(requestData as unknown as void, {
```
Comment (knagurski) [inline src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx:396]: I've never seen an instance where an arg was required to be `void`. Weird, but looks like the Service needs it ¯\_(ツ)_/¯

Comment (vkbansal) [reply src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx:396]: Yeah they are some type issue with `restful-react`

File: src/modules/70-pipeline/utils/useSaveTemplate.ts [hunk 4]
```diff
@@ -109,20 +109,13 @@ export function useSaveTemplate(TemplateContextMetadata: TemplateContextMetadata
...
-          JSON.stringify({
-            template: sanitize(temp, {
-              removeEmptyString: false,
-              removeEmptyObject: false,
-              removeEmptyArray: false
-            })
-          })
-        ),
-        {
-          version: '1.1'
```
Comment (knagurski) [inline src/modules/70-pipeline/utils/useSaveTemplate.ts:123]: Is this no longer needed?

File: src/modules/70-pipeline/utils/useSaveTemplate.ts [hunk 5]
```diff
@@ -109,20 +109,13 @@ export function useSaveTemplate(TemplateContextMetadata: TemplateContextMetadata
   const stringifyTemplate = React.useCallback(
     // Important to sanitize the final template to avoid sending null values as it fails schema validation
     (temp: NGTemplateInfoConfig) =>
-      yamlStringify(
-        JSON.parse(
-          JSON.stringify({
```
Comment (knagurski) [inline src/modules/70-pipeline/utils/useSaveTemplate.ts:114]: Do we know why this was here? I've seen similar techniques used to kill off non-primitive types. Is this no longer required?

**Review by knagurski — APPROVED** [review]
Just a question around a change to a weird bit of code, but apart from that, excellent job, looks very thorough!

File: src/modules/70-pipeline/utils/useSaveTemplate.ts (hunk 4 above)
Comment (vkbansal) [reply src/modules/70-pipeline/utils/useSaveTemplate.ts:123]: Not needed as ja-yaml handles this out of the box

File: src/modules/10-common/utils/YamlHelperMethods.ts (hunk 1 above)
Comment (knagurski) [reply src/modules/10-common/utils/YamlHelperMethods.ts:13]: Not a bad technique. I think our Sonar setup will complain, but it's a decent use-case 👍🏻

File: src/modules/70-pipeline/utils/useSaveTemplate.ts (hunk 5 above)
Comment (vkbansal) [reply src/modules/70-pipeline/utils/useSaveTemplate.ts:114]: It was needed due to https://github.com/eemeli/yaml/issues/239

**Review by deepesh-ui — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Review by jenil-harness — APPROVED** [review]

**Pull request merged**